server/mediaserver.go

```diff
@@ -72,6 +72,12 @@ type rtmpConnection struct {
 	// following fields should be protected by `lock`
 	sess *BroadcastSession
 	lock *sync.RWMutex
+
+	// Stuff for analytics
```

Comment:
Is there a way we can keep this self-contained within analytics? Eg, defining some struct within `monitor` and using the nonce to index into a map inside the `monitor` package. Then we can adjust counters as needed within specific events, eg `LogSegmentEmerged` and `LogSegmentTranscodeFailed`. I'd really like to avoid sprawling analytics code around various parts of the codebase if possible.

Also, I think most of these would need to be protected by a lock to make the race detector happy.